quadhoral-openmeteo/quadhoral_transform.py

```diff
@@ -0,0 +1,51 @@
+'''Transforms data obtained from openmeteo ready for loading'''
+import logging
+from datetime import datetime as dt
+
+
+def format_time(event_time: str) -> dt:
+    '''Returns a datetime object from a string'''
+    return dt.strptime(event_time, '%Y-%m-%dT%H:%M')
+
+def clean_data(county_data: dict) -> dict:
+    '''Returns a dictionary of weather data'''
+    return {'time': county_data['hourly']['time'],
+            'temp': county_data['hourly']['temperature_2m'],
+            'prob': county_data['hourly']['precipitation_probability'],
+            'precip': county_data['hourly']['precipitation'],
+            'clouds': county_data['hourly']['cloud_cover'],
+            'visibility': county_data['hourly']['visibility']}
+
+def format_data(clean_county_data: dict, county_id: int) -> list[list]:
+    '''Returns a list of lists containing county_id and weather data'''
+    if not clean_county_data['time']:
+        return []
+
+    return [[county_id, format_time(clean_county_data['time'][i]),
+             clean_county_data['temp'][i], clean_county_data['prob'][i],
+             clean_county_data['precip'][i], clean_county_data['clouds'][i],
+             clean_county_data['visibility'][i]] for i in range(24)]
+
+def transform(data: list[dict]) -> list[list]:
+    '''Orchestrates transform steps of pipeline'''
+    if not data:
+        raise ValueError('No data to transform.')
+
+    formatted_data = []
+    logging.info('Data transformation begun.')
```

Comment:
Clear logging that provides records of significant steps of the transform script.